src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketSender.cs

```diff
@@ -66,7 +66,7 @@ private List<ArraySegment<byte>> GetBufferList(in ReadOnlySequence<byte> buffer)
 
             if (_bufferList == null)
             {
-                _bufferList = new List<ArraySegment<byte>>();
+                _bufferList = new List<ArraySegment<byte>>((int)buffer.Length);
```

Comment:
Could we revert this change? The usage is a little different from other changes (we're pre-allocating the list based on the number of spans in a sequence) and I worry there might be security implications to this.